src/codegen/debug.rs

```diff
@@ -445,25 +445,31 @@ impl<'ink> DebugBuilder<'ink> {
             inkwell::AddressSpace::from(ADDRESS_SPACE_GLOBAL),
         );
 
-        // Handle auto-dereferencing pointers by creating a typedef if needed. This ensures
-        // that the DWARF information accurately reflects the intended usage (deref semantics) of the pointer type.
-        let ty = if let DataTypeInformation::Pointer { auto_deref: Some(_), .. } =
-            index.get_type(name).map(|it| it.get_type_information())?
-        {
-            let file = self.compile_unit.get_file();
-            let typedef_name = format!("__AUTO_DEREF__{name}");
-            self.debug_info.create_typedef(
-                pointer_type.as_type(),
-                &typedef_name,
-                file,
-                0, // Line 0 for built-in types
-                file.as_debug_info_scope(),
-                align_bits,
-            )
-        } else {
-            pointer_type
+        // For pointer types, we create a typedef to represent the pointer type in DWARF.
+        // This allows other tools to distinguish between different kinds of pointer semantics (e.g., reference vs. auto-deref, type-safety, ...)
+        // XXX: This is a workaround - gdb will show the mangled type name in the debugger, which is not ideal
+        let typedef_name = match index.get_type(name).map(|it| it.get_type_information())? {
+            DataTypeInformation::Pointer { auto_deref: Some(auto_deref), .. } => {
+                matches!(auto_deref, plc_ast::ast::AutoDerefType::Reference)
```

Comment:
iirc `AutoDerefType::Alias` is an alias for `AutoDerefType::Reference`, i.e. `foo AT bar: DINT` is syntactic sugar for `foo : REFERENCE TO DINT`. Not sure if you want to categorize both these enum variants as `__REFERENCE_TO`? Also is the case distinction necessary? Could all auto-dereference pointers be type-defed as `__AUTO_DEREF__{name}`?

Reply:
I'll have a look at the alias/reference to distinctions. Unfortunately, I need `REFERENCE TO` to have different semantics to aliases or `VAR_IN_OUT` variables, since they are required to show the `REFERENCE TO` in the type name, but var_in_out/alias auto-deref-pointers being pointers is an implementation detail and they should show their intrinsic type.